### ISSUE TITLE
Feature: realistic compass using

### ISSUE THREAD

**Issue by kOepi123** [KoffeinFlummi/AGM#2137]
Hello everyone,

through this I would like to request a realistic usage of the compass. I imagine two ways to do this:

1. make the compass a weapon, therefore it would be needed to be pulled like a pistol or similar.

2. attach a script to the compass ( I try to describe it the best I can ):

if player "agm_actionopencompass", 
             then "actionlowerweapon",
                         delay 1.5;
                                  execute "bis_actionopencompass";

(depending on toogle or hold binding)

if player "agm_actionclosecompass",
               then delay 1.5;

Atm you can use the compass like its burned into your brain, you push a button, it immediately appears and depending the way you are facing it might take some time until it show a correct azimuth. 

Pulling out a compass ( from slung way to wear ), opening, aiming, closing, letting it fall and taking a rifle into the hand, takes about 3 seconds ( average ) in real life without the balancing time it needs to give a correct bearing.

That is not realistic, because you will need both hands to operate a compass, which you cannot use them for holding your weapon ( arma 2 pulling binocular/rifle animation was outstanding ) anymore.

what would this change?

this would support better/more efficient communication for callouts and help to prevent the flow of uneccessarily detailed information.
in short:

you will only use the compass when you have the time to do it and when there is no other way to pass the information along.

let me know if I can help you with anything.

greetz

kOepi

**Comment by Raspu86** [comment]
I was thinking about opening an issue about the very same subject for quite some time now but I did not have the balls to do it. I know most of the arma community uses the (very very very) unrealistic way of calling out targets by compass bearing. I'd really love to see a delay implemented by agm!

**Comment by jonpas** [comment]
@Raspu86 never fear! Everyone involved in AGM loves to hear more and more, so don't be afraid and post, as long as it's something constructive. ;)

**Comment by esteldunedain** [comment]
We discussed something like this in the past. Here are my thoughts (shared by others at AGM):

1. Sure, making the compass some sort of weapon you can put in your hand would be the best solution (a la DayZ SA). I don't think it's possible though. The closest would probably be making it a pistol, but I think the hand animation cannot be modified to properly hold the compass. It's the same problem preventing treating the detonator from being modelled as a pistol (https://github.com/KoffeinFlummi/AGM/issues/1361#issuecomment-59828051)

2. Introducing a delay is probably possible, but it would feel completely wird without an animation conveying WHY are ou waiting. We don't have any suitable anim.

3. Requiring the weapon from being lowered for the compass to display would work, but it's far from perfect.

All this lead to:

4. From a broader perspective, inside the sim it's much more difficult to maintain spatial awareness than IRL. So is both spotting objects and pointing them to nearby friends. Having the compass easily accessible helps mitigate those problems a bit, while itself not being incredibly realistic. That's why I like STHUD too, which isn't realistic either, but partially compensates for the limitations of the virtual environment.

**Comment by kOepi123** [comment]
@esteldunedain

2. if you cannot find a suitable anim ( its only 1.5 seconds anim needed ), you could insert a text line, but even without I would prefer it and turn it on or off through a module, so people who dont want it, can disable it.

5. it is more difficult IRL, sure. I got nothing against the ability to use a compass, in some occasions it is a musthave ( reference points/navigation/anything that is further away than like 500 meters and stationary ).
STHUD is one way to compensate situational awareness, but by far not the fastest and most reliable one, e.g. it will not sustain a close quarters PvP match and it limits the needed training to "play as a team".

I really love the DayZ version having a finger to point and a compass as a weapon.

**Comment by BaerMitUmlaut** [comment]
>I know most of the arma community uses the (very very very) unrealistic way of calling out targets by compass bearing

You forget that pointing and gesturizing doesn't really work in Arma, and many house models look very similar (especially on Altis) or are visible multiple times. There's just some limits in calling out targets in Arma...

**Comment by kOepi123** [comment]
@BaermitUmlaut

IRL many house models look similar and even if not 100% identical, it is difficult to describe them with the tiny differences that they have. But we dont want to paint that house, so we only need the necessary details. That is why the only limits are the ones that you make yourself or the limit of your own vocabulary and your ability for abstract thinking ( Abstraktes Denken ).

Therefore you have some Tools IRL, "3 Finger Method", "Reference points", "Target Reference Points", "Arc referring", "The 3Ds". and when I continue thinking I will find more ways to not use a compass or a map. And most of them you can use in Arma already.

**Comment by KoffeinFlummi** [comment]
I have to agree with @esteldunedain here; While getting your compass out that fast is not realistic, it is necessary to compensate for the lack of spatial and situational awareness compared to real life.

**Comment by kOepi123** [comment]
looks like you guys cannot play without the compass. I would love to take you out on a no-compass session.
I guess its the same as people once said about not having GPS, not having friendly markers in map or in direct, 3rd person view. what all these things have in common besides being unrealistic is that you have to overcome it with realistic communication and attention.

 as I said there are other ways to communicate among and identify targets up to distance of roughly 300 meters. and for anything further I can see why you would take out a compass to identify more accurately.
atm the problem is not only for the one that calls out the target, but especially for those who recieve that callout, taking out their compasses and trying to find that target while focussing on their compass only instead of their environment.

There is the clocksystem ( 1 to 12 oclock ) and general direction system ( front right to front ) which can be applied more efficiently, as the group knows which direction the group is moving and depending on that you can call out any direction and in that instant you hear it, you only need to transform it in your brain and then know where to look at, which is way faster.
If the target is not that close, you can use buildings or anything else as references or even reference points in built up areas. So according to the 3Ds you would say.

"Contact, Front Right, 300 meters, behind the two-story (infantry)" ( if the building would be obvious)

while the callout with a compass would be.

Sorry to spam you here.

"Contact, Zero Four Five ( 300 meters )."
that way is shorter and more accurate in the callout itself, but unrealistically fast determined, plus
the recievers of the callout have a delay of opening the compass ( unless you run around with your compass open all the time. lol ).

within these short ranges of roughly 400 meters the time saving is more important than accuaracy, while with longer distances you have more time and the accuracy becomes more important.

Sorry to spam you.

**Comment by BaerMitUmlaut** [comment]
Not everybody will be facing the same way, and you don't necessarily see what way other people are looking because your FOV is way lower than irl. That's why the clock system or saying front or right will not always work. Saying "south-east" etc. on the other hand will work and isn't that unrealistic in my opinion. If you don't want people to use the compass, remove it from your loadout. If you want them to use the compass less, train them to call out targets without it.

A progress bar that just delays the compass showing up breaks immersion and doesn't really add anything in my opinion.

**Comment by kOepi123** [comment]
@BaerMitUmlaut 
I tested the FOV in arma 2 in a sample mission and it was way smaller. 
I tested it in arma 3 alpha and it was exactly as expected.
I would be interested in finding out again about it, do you have a link or something which shows people testing it?

My problem is that I cannot simply remove the compass, because the sun is not standing correctly in the sky and for some things it is very difficult not to use a compass and IRL everyone gets issued a compass, but not for calling out targets.

**Comment by BaerMitUmlaut** [comment]
Real life "FOV" is 180°, Arma 3s is around 100 (on my screen, 1080p, 16:9).

Again, if you want them to use the compass less, train them to call out targets without it. Just making a delay to open the compass is not really going to change the way they call out targets.

**Comment by kOepi123** [comment]
yes, it will change the way they are getting used to it. it would support that method of calling it out.
And as I think it would support everyone playing this game, so I suggested this thread. As a module would make it possible to turn it on and off when desired or not, it would give everyone the option to play as they prefer it.
I am not out on workarounds.

how do you measure the FOV?

**Comment by BaerMitUmlaut** [comment]
>how do you measure the FOV?

Get to distinct points on the border of your screen (I used trees) so that they are just visible. Measure the bearing to both, the difference between the bearings is your FOV.

**Comment by kOepi123** [comment]
got it. so if you would be changing your FOV to near 180 °, you would agree with me?

just trying it out with the FOV changer ...

it does not need everyone to face the same direction to use the clocksystem. It is always depending on the groups movement, not determined by the one who is calling it out ( wrong since ofp ).

**Comment by bux578** [comment]
> Not everybody will be facing the same way, and you don't necessarily see what way other people are looking because your FOV is way lower than irl.

IRL it's about the movement of the group, not the individual person. (just my two cents ... *fliesaway*)

**Comment by BaerMitUmlaut** [comment]
>so if you would be changing your FOV to near 180 °, you would agree with me?

Sure. But 180° FOV is unplayable with one monitor.

**Comment by vackosar** [comment]
I am also annoyed by unrealism of compass in arma. Compass slowing mod would help reducing nonsensical compass overuse in coops. Limitness of monitors dont justify this. It would be funny to see airsofter running around with compass on their necks constantly using it.

**Comment by BaerMitUmlaut** [comment]
@vackosar: Brah, AGM is dead since almost a year now.

**Comment by kOepi123** [comment]
thank you for your input vackosar. Your attitude shows me again how ridiciolous it is to have this kind of compass use. This thread is dead though, as is AGM.

**Comment by vackosar** [comment]
@kOepi123, glad we agree. I submitted another request into ACE3: https://github.com/acemod/ACE3/issues/414#issuecomment-195998695
Ironically someone requested on ACE3 to see compass when in sighting mode recently: https://github.com/acemod/ACE3/issues/414#issuecomment-120169828

**Comment by kOepi123** [comment]
@vackosar 
Atm it is not about realism, it is more about the question, how to implement it contrary those opinions from the people that can implement it ( like 3 or 4 guys in this thread ) and more importantly how to get the resources necessary to implement it.
So convincing and making it possible to implement is here more important than the realism argument.
There are many ways to work around this issue and it would change the communication heavily when this is going active, so there need to be good arguments on why there should not be a work around, but a technical limitation, even when optional as a module.

So the first goal to achieve is to show how it is done IRL, how it can be done in the game and what it would need to accomplish it. a typical ACE development. First nobody wants it, then nobody knows how work it, then people learn it, then they cant without it.

**Comment by vackosar** [comment]
Well the realism is kind of Arma's selling point. Maybe @Dsylecxi would help: https://www.youtube.com/watch?v=v7zoVIsIT2A

**Comment by kOepi123** [comment]
well vanilla arma is not really a lot more realistic than another shooter. The big impact have the mods and therefore the community that strive for realism.
Dsylecxi on the other hand even has the sthud mod running to overcome the so called "lack of situational awareness". So it wont be in his interest to join our aim.
I wouldnt invest any time to try to convince a celebrity that openly goes a different direction of realism to convince other people to bring more realism.

**Comment by aster-isco** [comment]
I am looking for the same thing. I made a script that disorients the compass when you have a weapon in your hand, or when you are inside or very close to a vehicle (in this case, a car. But it can be a tank or any object). I was unable to add a delay to access the compass. I'm not a programmer, so there must be a better way to do this.

`while {true} do {
	if(
		visibleCompass and
		!(currentWeapon player == "")
		or !isNull objectParent player
	)then{
		setCompassOscillation [rad random [5, 60, 80], 0.1, 1];
	}else{
			_obj_metalico = nearestObject [player, "Car"];
      if(
        player distance _obj_metalico <= 5
      )then{
        setCompassOscillation [rad random [5, 60, 80], 0.1, 1];
      }else{
        setCompassOscillation [-1, 0, 0];
      };
	};
	sleep 2;
}`

**Comment by BaerMitUmlaut** [comment]
AGM has closed down 6 years ago. What are you doing here?

**Comment by PabstMirror** [comment]
Realistic magnetic-declination for each map would also be an interesting feature to add
